packages/ragbits-core/src/ragbits/core/vector_stores/_cli.py

```diff
@@ -0,0 +1,117 @@
+import asyncio
+from dataclasses import dataclass
+from pathlib import Path
+
+import typer
+from pydantic import BaseModel
+from rich.console import Console
+
+from ragbits.cli import cli_state, print_output
+from ragbits.cli.state import OutputType
+from ragbits.core.config import core_config
+from ragbits.core.embeddings.base import Embeddings
+from ragbits.core.utils.config_handling import InvalidConfigError
+from ragbits.core.vector_stores.base import VectorStore, VectorStoreOptions
+
+vector_stores_app = typer.Typer(no_args_is_help=True)
+
+
+@dataclass
+class CLIState:
+    vector_store: VectorStore | None = None
+
+
+state: CLIState = CLIState()
+
+
+@vector_stores_app.callback()
+def common_args(
+    factory_path: str | None = None,
+    yaml_path: str | None = None,
+) -> None:
+    try:
+        state.vector_store = VectorStore.subclass_from_defaults(
+            core_config,
+            factory_path_override=factory_path,
+            yaml_path_override=Path.cwd() / yaml_path if yaml_path else None,
+        )
+    except InvalidConfigError as e:
+        Console(stderr=True).print(e)
+        raise typer.Exit(1) from e
+
+
+@vector_stores_app.command(name="list")
+def list_entries(limit: int = 10, offset: int = 0) -> None:
+    """
+    List all objects in the chosen vector store.
+    """
+
+    async def run() -> None:
+        if state.vector_store is None:
+            raise ValueError("Vector store not initialized")
+
+        entries = await state.vector_store.list(limit=limit, offset=offset)
+        print_output(entries)
+
+    asyncio.run(run())
+
+
+class RemovedItem(BaseModel):
+    id: str
+
+
+@vector_stores_app.command()
+def remove(ids: list[str]) -> None:
+    """
+    Remove objects from the chosen vector store.
+    """
+
+    async def run() -> None:
+        if state.vector_store is None:
+            raise ValueError("Vector store not initialized")
+
+        await state.vector_store.remove(ids)
+        if cli_state.output_type == OutputType.text:
+            typer.echo(f"Removed entries with IDs: {', '.join(ids)}")
+        else:
+            print_output([RemovedItem(id=id) for id in ids])
+
+    asyncio.run(run())
+
+
+@vector_stores_app.command()
+def query(
+    text: str,
+    k: int = 5,
+    max_distance: float | None = None,
+    embedder_factory_path: str | None = None,
+    embedder_yaml_path: str | None = None,
+) -> None:
+    """
+    Query the chosen vector store.
+    """
+
+    async def run() -> None:
```

Comment:
here also no concurrent exeution within the  task , however thought that implementation of batched query on list of texts should be quiet straightforward

Reply:
however maybe let's create backlog ticket for that

Reply:
I don't understand what could be made concurrent here

Reply:
i mean that we could take a list of texts instead of single text as an input and write asynchronous nested run which takes text as an argument and process all the text concurrently - eventually which some batch size that could be another parameter. think that it is quiet low hanging fruit

Reply:
Remember that this is a part of a CLI interface, so for humans to easily check stuff, and with its specific limitations. There is no data structures per se - just command line arguments as input, and writing text to standard output as output.

Right now one would use it like that:
```
ragbits vectore-store query "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas ac gravida justo, at venenatis neque. Nullam lacinia varius auctor. Praesent quis pretium augue."
```

And the output is a table of stuff from the data-store similar to the given text.

Of course I can change it so you can give an arbitrary number of texts separated by spaces, like this (although for me personally this doesn't look user-friendly):

```
ragbits vectore-store query "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas ac gravida justo, at venenatis neque. Nullam lacinia varius auctor. Praesent quis pretium augue." "raesent ante diam, elementum ut ligula nec, mattis vestibulum tellus. Etiam euismod, odio a varius interdum, arcu urna faucibus lorem, eu rhoncus sapien neque ut purus. " "Nulla facilisi. Interdum et malesuada fames ac ante ipsum primis in faucibus."
```

But what would be the output? Multiple tables - one table for each input text? That's not something that `print_output` currently supports, and I also think that would be confusing.

Reply:
i would add task to discuss o grooming